### PR TITLE
Add extensions di snippets to Af, Unity and SM

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

File: Snippets/Castle/Extensions.DependencyInjection_1/Usage.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class Usage
+{
+    Usage(EndpointConfiguration endpointConfiguration)
+    {
+        #region CastleWindsor
+
+        var containerSettings = endpointConfiguration.UseContainer(new WindsorServiceProviderFactory());
+
+        containerSettings.ConfigureContainer(c => c.Register(Component.For<MyService>()
+                .Instance(new MyService())));
```
Comment (timbussmann) [inline Snippets/Castle/Extensions.DependencyInjection_1/Usage.cs]: the line break is a bit weird here imo. I'd suggest moving the registration part onto a single line

File: Snippets/StructureMap/Extensions.DependencyInjection_1/Extensions.DependencyInjection_1.csproj [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net462</TargetFramework>
+    <LangVersion>7.1</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus.Extensions.DependencyInjection" Version="1.*" />
+    <PackageReference Include="StructureMap.Microsoft.DependencyInjection" Version="2.0.0" />
```
Comment (timbussmann) [inline Snippets/StructureMap/Extensions.DependencyInjection_1/Extensions.DependencyInjection_1.csproj]:
```suggestion
    <PackageReference Include="StructureMap.Microsoft.DependencyInjection" Version="2.*" />
```

should this use a wildcard dependency too?

File: Snippets/Unity/Extensions.DependencyInjection_1/Extensions.DependencyInjection_1.csproj [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <LangVersion>7.1</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus.Extensions.DependencyInjection" Version="1.*" />
+    <PackageReference Include="Unity.Microsoft.DependencyInjection" Version="5.11.5" />
```
Comment (timbussmann) [inline Snippets/Unity/Extensions.DependencyInjection_1/Extensions.DependencyInjection_1.csproj]: are we more careful with unity dependencies in the snippets too due to our versioning policy with Unity? https://docs.particular.net/nservicebus/dependency-injection/unity#breaking-changes-in-unity

Comment (andreasohlund) [reply Snippets/Unity/Extensions.DependencyInjection_1/Extensions.DependencyInjection_1.csproj]: Given the stability of the MS DI seam I think we can go with a wildcard, I will change

**Pull request merged**